qualtran/bloqs/generalized_qsp.py

```diff
@@ -63,15 +71,12 @@ def rotation_matrix(self):
             ]
         )
 
-    def decompose_from_registers(
-        self, *, context: cirq.DecompositionContext, q: NDArray[cirq.Qid]
-    ) -> cirq.OP_TREE:
-        qubit = q[0]
-
-        yield cirq.Rz(rads=np.pi - self.lambd).on(qubit)
-        yield cirq.Ry(rads=2 * self.theta).on(qubit)
-        yield cirq.Rz(rads=-self.phi).on(qubit)
-        yield cirq.GlobalPhaseGate(np.exp(1j * (np.pi + self.lambd + self.phi) / 2)).on()
+    def build_composite_bloq(self, bb: 'BloqBuilder', q: 'SoquetT') -> Dict[str, 'SoquetT']:
+        q = bb.add(ZPowGate(exponent=2, global_shift=0.5), q=q)
+        q = bb.add(ZPowGate(exponent=1 - self.lambd / np.pi, global_shift=-1), q=q)
+        q = bb.add(Ry(angle=2 * self.theta), q=q)
+        q = bb.add(ZPowGate(exponent=-self.phi / np.pi, global_shift=-1), q=q)
+        return {'q': q}
```

Comment:
Yep it works now (inheriting from GWR). But I thought it was qualtran gates/bloqs are preferred to cirq ones?

Reply:
The good thing about `basic_gates/` is that we now do an automatic conversion from Cirq gates to Qualtran Bloqs wherever applicable. So your earlier implementation (deriving from `GWR`) would (a) Use the Cirq native gates when using a cirq-style API for simulations, which is often faster (see https://github.com/quantumlib/Qualtran/issues/793) and (b) Use the qualtran-native Bloqs when accessed via qualtran API like `bloq.decompose_bloq()` or `bloq.call_graph()`. 

Since this is being moved to `basic_gates/`, having a qualtran-only API (like you have right now) is good and an explicit conversion to/from `PhasedXZGate` in Cirq would be nice.